enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/identity/ClusterBinder.java

```diff
@@ -111,6 +113,20 @@ else if ( !coreTopology.canBeBootstrapped() )
      */
     public BoundState bindToCluster() throws Throwable
     {
+
+        if ( dbNameStorage.exists() )
+        {
+            String storedName = dbNameStorage.readState().name();
+            if ( !dbName.equals( storedName ) )
+            {
+                throw new IllegalStateException( format("Your configured database name may have changed. Found %s, expected %s", dbName, storedName ) );
+            }
+        }
+        else
+        {
+            dbNameStorage.writeState( new DatabaseName( dbName ) );
+        }
```

Comment:
Maybe this block would be better as a method on `SimpleStorage`?

Reply:
Not sure what you mean?

Reply:
Lines 117 to 128 dealing with checking the existing storage, setting if its empty and validating if it's not. These could be in a method on `SimpleStorage`, or some helper class if we want to keep `SimpleStorage` as simple as possible.